exec/vm.go

```diff
@@ -69,7 +71,7 @@ var endianess = binary.LittleEndian
 
 // NewVM creates a new VM from a given module. If the module defines a
 // start function, it will be executed.
-func NewVM(module *wasm.Module) (*VM, error) {
+func NewVM(module *wasm.Module, recoverExecutionPanic bool) (*VM, error) {
```

Comment:
I suspect you shouldn't add the `recoverExecutionPanic` field. Most users wouldn't need to set it, so having it be mandatory here would just make the API harder to use. It's not documented here, so users won't know what it does.

Instead, users who want to have `ExecCode` recover from panics can do:

```Go
vm, err := wagon.NewVM(m)
if err != nil {
    // handle error
}
vm.RecoverExecutionPanic = true
```

Reply:
Alright, I'll remove the option. variadic options are nice and cool, however I feel introducing them should be its own PR.